File: TableGames.Games.Doubt/PlayerCup.cs
using System.Collections.Generic;
using System.Linq;
using TableGames.Domain;

namespace TableGames.Games.Doubt
{
    public enum LockStatus
    {
        Available,
        Locking,
        Unavailable
    }

    public class PlayerCup
    {
        public Player Player { get; set; }
        public List<Dice> Dices { get; set; }
        public LockStatus LockStatus { get; set; }

        public PlayerCup(Player player, PlayerBag playerBag) {
            Player = player;
            Dices = new List<Dice>(Enumerable.Range(1, playerBag.DicesQuantity).Select(i => new Dice(true)));
            LockStatus = playerBag.LockStatus;
        }

        public void ExposeDices() {
            Dices.ForEach(d => d.IsExposed = true);
        }

        public object ToClient() {
            return new {
                playerName = Player.Name,
                dices = Dices.Select(d => d.ToClient()),
                lockStatus = LockStatus.ToString()
            };
        }
    }
}

File: TableGames.Games.Doubt/playerCup.js
define(['game!Doubt/dice'], function(Dice) {

    function PlayerCup(playerCupState, doubt) {
        var self = this;
        self.playerName = playerCupState.playerName;
        self.dices = playerCupState.dices.map(function(diceState) { return new Dice(diceState, doubt); });
        self.lockStatus = playerCupState.lockStatus;

        self.updateDices = function(dicesState) {
            dicesState.forEach(function(diceState, index) {
                self.dices[index].value(diceState.value);
                self.dices[index].isExposed(diceState.isExposed);
            });
        };
    }

    return PlayerCup;
});
